server_util_test.go

```diff
@@ -37,8 +39,10 @@ type server struct {
 
 type handler func(ctx context.Context, args *raw.Args) (*raw.Res, error)
 
-func newServer(t *testing.T) *server {
-	ch := testutils.NewServer(t, testutils.NewOpts().SetServiceName("foo").DisableLogVerification())
+func newServer(t *testing.T, tracer opentracing.Tracer) *server {
```

Comment:
Rather than take a tracer (and get into the pattern of everyone passing `nil` for default options), we should probably just implement a lightweight options pattern here,


```go
func withTracer(opentracing.Tracer) func(*testutils.ChannelOpts) {
  return func(opts *testutils.ChannelOpts) {
    opts.Tracer = tracer
  }
}

func newServer(t *testing.T, opts... func(*testutils.ChannelOpts)) {
  chOpts := testutils.NewOpts().[...]
  for _, opt := range opts {
    opt(chOpts)
  }
  [...]
}
```